themes-default/slim/src/store/modules/shows.js

```diff
@@ -81,20 +81,32 @@ const actions = {
      * @param {ShowParameteres[]} shows - Shows to get. If not provided, gets the first 10k shows.
      * @returns {(undefined|Promise)} undefined if `shows` was provided or the API response if not.
      */
-    getShows(context, shows) {
+    async getShows(context, shows) {
         const { commit, dispatch } = context;
 
-        // If no shows are provided get the first 10k
+        // If no shows are provided get the first 10000
         if (!shows) {
+            const limit = 1000;
+            const page = 1;
+
             const params = {
-                limit: 10000
+                limit,
+                page
             };
-            return api.get('/series', { params }).then(res => {
-                const shows = res.data;
-                return shows.forEach(show => {
-                    commit(ADD_SHOW, show);
-                });
-            });
+
+            const records = [];
+            let keepGoing = true;
+            while (keepGoing) {
+                const response = await api.get('/series', { params });
```

Comment:
I'd suggest instead look at the header you get back as it should have the pagination info in it. Instead of doing the while loop try this. The header should also have the total pages so you can then easily know the end result instead of relying on a boolean being toggled.

- Get Page 1
- `commit(ADD_SHOW, show)`
- Check pagination header
- call `api.get` again with new page param
- rinse and repeat